core/src/apps/management/recovery_device/homescreen.py

```diff
@@ -142,6 +195,27 @@ async def _finish_recovery_dry_run(secret: bytes, backup_type: BackupType) -> Su
         raise wire.ProcessError("The seed does not match the one in the device")
 
 
+async def _finish_recovery_unlock_repeated_backup(
+    secret: bytes, backup_type: BackupType
+) -> Success:
+    from apps.common import backup
+
+    if backup_type is None:
+        raise RuntimeError
+
+    is_slip39 = backup_types.is_slip39_backup_type(backup_type)
+
+    result = _check_secret_against_stored_secret(secret, is_slip39, backup_type)
+
+    if result:
+        backup.activate_repeated_backup()
+        storage_recovery.end_progress()
+        return Success(message="Backup unlocked")
+    else:
+        storage_recovery.end_progress()
```

Comment:
move to the common part before `if result`

Reply:
https://github.com/trezor/trezor-firmware/pull/3743/commits/5552105eca7966328c6def0def4f2a62b5233b94#diff-911c38bc5c90da24b97abd99eecd2816097835d2166c6ec6dd0755a78566f3adR210